overlord/state/taskrunner.go

```diff
@@ -260,7 +260,8 @@ func (r *TaskRunner) run(t *Task) {
 			case UndoneStatus:
 				next = t.WaitTasks()
 			}
-			if len(next) > 0 {
+			chg := t.Change()
+			if len(next) > 0 || (chg != nil && chg.IsReady()) {
 				r.state.EnsureBefore(0)
 			}
```

Comment:
Hm shouldn't this be `if len(next) > 0 || (chg != nil && chg.IsReady() && !chg.IsClean())` ?

Reply:
And probably could use a comment:
```
// trigger ensure so that the next ready task can start, or the task cleanups can run next
```